scipy/stats/tests/test_multivariate.py

```diff
@@ -1860,18 +1860,25 @@ def test_det_and_ortho(self):
         dets = np.array([[np.linalg.det(x) for x in xx] for xx in xs])
         assert_allclose(np.fabs(dets), np.ones(dets.shape), rtol=1e-13)
 
-        # Test that we get both positive and negative determinants
-        # Check that we have at least one and less than 10 negative dets in a sample of 10. The rest are positive by the previous test.
-        # Test each dimension separately
-        assert_array_less([0]*10, [np.nonzero(d < 0)[0].shape[0] for d in dets])
-        assert_array_less([np.nonzero(d < 0)[0].shape[0] for d in dets], [10]*10)
-
         # Test that these are orthogonal matrices
         for xx in xs:
             for x in xx:
                 assert_array_almost_equal(np.dot(x, x.T),
                                           np.eye(x.shape[0]))
 
+    @pytest.mark.parametrize("dim", [2, 5, 10, 20])
+    def test_det_distribution_gh18272(self, dim):
+        # Test that positive and negative determinants are equally likely.
+        rng = np.random.default_rng(6796248956179332344)
+        dist = ortho_group(dim=dim)
+        rvs = dist.rvs(size=5000, random_state=rng)
+        dets = np.linalg.det(rvs)
+        k = np.sum(dets > 0)
+        n = len(dets)
+        res = stats.binomtest(k, n)
+        low, high = res.proportion_ci(confidence_level=0.95)
+        assert low < 0.5 < high
```

Comment:
> It also means that checking the confidence interval is really no better than checking that the p-value of the binomial test is greater than 0.05. It isn't wrong to use the confidence interval, but it doesn't really provide any benefit in this test.

I know. I just preferred that way of expressing the idea. I didn't mind that it was slower because it's still fast.

```python3
%timeit stats.binomtest(k, n)
1.18 ms ± 35.8 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

%timeit res.proportion_ci(confidence_level=0.95)
2.83 ms ± 10.2 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

But if you prefer, feel free to commit:

```suggestion
        assert res.pvalue > 0.05
```